aries_vcx/tests/test_pgsql_plugin.rs

```diff
@@ -0,0 +1,117 @@
+            Ok(())
+        }
+    }
+}
+
+#[derive(Clone, Debug, Deserialize, Serialize)]
+pub struct WalletStorageConfig {
+    // Wallet storage type for agents wallets
+    #[serde(rename = "type")]
+    pub xtype: Option<String>,
+    // Optional to override default library path. Default value is determined based on value of
+    // xtype and OS
+    pub plugin_library_path: Option<String>,
+    // Optional to override default storage initialization function. Default value is  determined
+    // based on value of xtype and OS
+    pub plugin_init_function: Option<String>,
+    // Wallet storage config for agents wallets
+    pub config: Option<Value>,
+    // Wallet storage credentials for agents wallets
+    pub credentials: Option<Value>,
+}
+
+#[test]
+#[cfg(feature = "test_integration")]
+fn it_should_use_pgsql_wallet_to_provision_cloud_agent() {
+    let storage_config = r#"
+          {
+            "config": {
+              "url": "localhost:5432",
+              "max_connections" : 90,
+              "connection_timeout" : 30,
+              "wallet_scheme": "MultiWalletSingleTableSharedPool"
+            },
+            "credentials": {
+              "account": "postgres",
+              "password": "mysecretpassword",
+              "admin_acount": "postgres",
+              "admin_password": "mysecretpassword"
+            },
+            "type": "postgres_storage"
+          }"#;
+    let storage_config: WalletStorageConfig = serde_json::from_str(storage_config).unwrap();
+    info!("Init pgsql storage: Starting.");
+    _init_wallet(&storage_config).unwrap();
+    info!("Init pgsql storage: Finished.");
+    let storage_credentials = storage_config.credentials.map(|a| serde_json::to_string(&a).unwrap()).clone();
+    let storage_config = storage_config.config.map(|a| serde_json::to_string(&a).unwrap()).clone();
+
+    settings::clear_config();
+    let enterprise_seed = "000000000000000000000000Trustee1";
+    let config_wallet = WalletConfig {
+        wallet_name: format!("faber_wallet_{}", uuid::Uuid::new_v4().to_string()),
+        wallet_key: settings::DEFAULT_WALLET_KEY.into(),
+        wallet_key_derivation: settings::WALLET_KDF_RAW.into(),
+        wallet_type: Some("postgres_storage".into()),
+        storage_config,
+        storage_credentials,
+        rekey: None,
+        rekey_derivation_method: None,
+    };
+    let config_provision_agent = AgentProvisionConfig {
+        agency_did: AGENCY_DID.to_string(),
+        agency_verkey: AGENCY_VERKEY.to_string(),
+        agency_endpoint: AGENCY_ENDPOINT.to_string(),
+        agent_seed: None,
+    };
+    create_wallet(&config_wallet).unwrap();
+    open_as_main_wallet(&config_wallet).unwrap();
+    let config_issuer = configure_issuer_wallet(enterprise_seed).unwrap();
+    init_issuer_config(&config_issuer).unwrap();
+    provision_cloud_agent(&config_provision_agent).unwrap();
+    close_main_wallet().unwrap();
+    assert_eq!(4, 4);
+}
```

Comment:
I wonder whether the test necessarily belongs here. Maybe `aries_vcx::libindy::utils::wallet`? But again, you put the test-only code here...

Reply:
It's kind of the other way arround - our existing organization is somewhat incorrect. Integation tests belong to `test` directory, whilst unit tests should be directly in the files/modules they test.

https://doc.rust-lang.org/book/ch11-03-test-organization.html

> The Rust community thinks about tests in terms of two main categories: unit tests and integration tests. Integration tests are entirely external to your library and use your code in the same way any other external code would, using only the public interface and potentially exercising multiple modules per test.

> To create integration tests, you first need a tests directory.

This pgsql is integration test, so I think the placement is correct.

Reply:
But it is inconsistent until we move all integration tests :)

Reply:
Yes, so we should start moving :-) I think we can do that as part of `libvcx ---> aries-vcx` integration test migration